atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/service/PreStartHandlingTransactionService.java

```diff
@@ -96,27 +112,29 @@ public void close() {
         delegate.close();
     }
 
-    private ListenableFuture<Long> getFromDelegate(
-            long startTimestamp, AsyncTransactionService asyncTransactionService) {
+    private <T> ListenableFuture<T> getFromDelegate(
+            long startTimestamp, Function<Long, ListenableFuture<T>> getter, T invalidCommit) {
```

Comment:
nit: I'd name this `commitValueForInvalidStart` or something like that, for clarity.